tuning/trainercontroller/callback.py

```diff
@@ -235,6 +261,14 @@ def _take_control_actions(self, event_name: str, **kwargs):
                     raise NameError(
                         "Rule failed due to use of disallowed variables"
                     ) from en
+                except NameNotDefined as en1:
+                    raise NameError(
+                        "Rule failed because some of the variables are not defined"
```

Comment:
Are there any ways to make these exceptions more specific, e.g., what variables are not defined for this one, or what features are not supported in the one below this?

Reply:
Yes the Exception does contain the name of the variable that is not defined https://github.com/foundation-model-stack/fms-hf-tuning/pull/147/files#diff-73791d95640c88e27cc8aef08ee77265bd2939f7f7a0691640aabb3f6394d932R62

We can add more information if necessary.